assets/src/components/reader-theme-carousel/index.js

```diff
@@ -0,0 +1,170 @@
+							collection.shownThemes.push( theme );
+						}
+					} else {
+						collection.shownThemes.push( theme );
+					}
+
+					return collection;
+				},
+				{ shownThemes: [], hasUnavailableThemes: false },
+			),
+		[ currentTheme.name, hideCurrentlyActiveTheme, includeUnavailableThemes, themes ],
+	);
+
+	// Memoize carousel items to avoid flickering images on every render.
+	const carouselItems = useMemo(
+		() => shownThemes
+			.map( ( theme ) => (
+				{
+					label: theme.name,
+					name: theme.slug,
+					Item: () => (
+						<ThemeCard
+							disabled={ 'non-installable' === theme.availability }
+							ElementName="div"
+							screenshotUrl={ theme.screenshot_url }
+							{ ...theme }
+						/>
+					),
+				} ) ),
+		[ shownThemes ],
+	);
+
+	if ( fetchingThemes ) {
+		return <Loading />;
+	}
+
+	return (
+		<div className="reader-theme-selection">
+			<h3>
+				{ __( 'Choose Reader Theme', 'amp' ) }
+			</h3>
+			<p>
+				{
+					// @todo Probably improve this text.
+					__( 'Select the theme template for mobile visitors', 'amp' )
+				}
+			</p>
+			{ activeTheme && hideCurrentlyActiveTheme && (
+				<AMPNotice>
+					<p>
+						{
+							sprintf(
+								/* translators: placeholder is the name of a WordPress theme. */
+								__( 'Your active theme “%s” is not available as a reader theme. If you wish to use it, Transitional mode may be the best option for you.', 'amp' ),
+								activeTheme.name,
+							)
+						}
+					</p>
+				</AMPNotice>
+			) }
+			<div>
+				{ 0 < shownThemes.length && (
+					<Carousel
+						items={ carouselItems }
+						highlightedItemIndex={ shownThemes.findIndex( ( { name } ) => name === selectedTheme.name ) }
+					/>
+				) }
+
+				{
+					hasUnavailableThemes && (
+						<AMPNotice type={ NOTICE_TYPE_WARNING }>
+							<p>
+								{ AMP_QUERY_VAR_CUSTOMIZED_LATE
+								/* dangerouslySetInnerHTML reason: Injection of code tags. */
+									? <span
+										dangerouslySetInnerHTML={ {
+											__html: sprintf(
+											/* translators: 1: customized AMP query var, 2: default query var, 3: the AMP_QUERY_VAR constant name, 4: the amp_query_var filter, 5: the plugins_loaded action */
+												__( 'The following themes are not available because your site (probably the active theme) has customized the AMP query var too late (it is set to %1$s as opposed to the default of %2$s). Please make sure that any customizations done by defining the %3$s constant or adding an %4$s filter are done before the %5$s action with priority 8.', 'amp' ),
+												`<code>${ AMP_QUERY_VAR }</code>`,
+												`<code>${ DEFAULT_AMP_QUERY_VAR }</code>`,
+												'<code>AMP_QUERY_VAR</code>',
+												'<code>amp_query_var</code>',
+												'<code>plugins_loaded</code>',
+											),
+										} }
+									/>
+									: __( 'Some supported themes cannot be installed automatically on this site. To use, please install them manually or contact your hosting provider', 'amp' )
```

Comment:
```suggestion
									: __( 'Some supported themes cannot be installed automatically on this site. To use, please install them manually or contact your hosting provider.', 'amp' )
```

Reply:
Would the notice be better suited above the carousel?

![image](https://user-images.githubusercontent.com/16200219/88570984-7f3c0600-d02c-11ea-9649-131dd68deb7c.png)

Reply:
Done in faed83a22

Reply:
Also made that "Unavailable" black as was suggested earlier.